desktop/core/src/desktop/api2.py

```diff
@@ -520,13 +520,23 @@ def create_directory(request):
 
   parent_dir = Directory.objects.get_by_uuid(user=request.user, uuid=parent_uuid, perm_type='write')
 
-  directory = Directory.objects.create(name=name, owner=request.user, parent_directory=parent_dir)
-
-  return JsonResponse({
+  # Check if directory already exists then return it to avoid creating duplicate directory
+  try:
+    dir_exist = Directory.objects.get(name=name, owner=request.user, parent_directory=parent_dir)
```

Comment:
Use insted https://docs.djangoproject.com/en/3.1/ref/models/querysets/#get-or-create ?

`dir, created = get_or_create()`

Reply:
(actually better to not change it as "probably" not related to problem)